extensions/wikia/ContentReview/services/ContentReviewStatusesService.class.php

```diff
@@ -0,0 +1,324 @@
+			case 3:
+			case 5: $statusKey = self::STATUS_APPROVED; break;
+			case 4: $statusKey = self::STATUS_REJECTED; break;
+			default: $statusKey = self::STATUS_NONE;
+		}
+
+		return $statusKey;
+	}
+
+	/**
+	 * Gets live revision id
+	 *
+	 * Live revision is revision which was last approved or auto approved
+	 * Returns 0 if there is no live revision
+	 *
+	 * @param array $pageStatuses
+	 * @return int
+	 */
+	public function getLiveRevision( $pageStatuses ) {
+		if ( isset( $pageStatuses[ReviewModel::CONTENT_REVIEW_STATUS_APPROVED] ) ) {
+			return $pageStatuses[ReviewModel::CONTENT_REVIEW_STATUS_APPROVED];
+		}
+
+		if ( isset( $pageStatuses[ReviewModel::CONTENT_REVIEW_STATUS_AUTOAPPROVED] ) ) {
+			return $pageStatuses[ReviewModel::CONTENT_REVIEW_STATUS_AUTOAPPROVED];
+		}
+
+		return 0;
+	}
+
+	/**
+	 * Prepares params for diff revision link
+	 *
+	 * @param int $revisionId
+	 * @param int $liveRevisionId
+	 * @return array
+	 */
+	public function prepareRevisionLinkParams( $revisionId, $liveRevisionId = 0 ) {
+		$params = [];
+
+		if ( !empty( $liveRevisionId ) ) {
+			$params['oldid'] = $liveRevisionId;
+
+			if ( $revisionId !== $liveRevisionId ) {
+				$params['diff'] = $revisionId;
+			}
+		} else {
+			$params['oldid'] = $revisionId;
+		}
+
+		return $params;
+	}
+
+	public static function purgeJsPagesCache() {
+		\WikiaDataAccess::cachePurge( self::getJsPagesMemcKey() );
+	}
+
+	protected function initPageData( \Title $title ) {
+		$statusNoneMsg = $this->getStatusMessage();
+
+		return [
+			'pageLink' => $this->createPageLink( $title ),
+			'pageName' => $title->getFullText(),
+			'latestRevision' => [
+				'statusKey' => self::STATUS_NONE,
+				'message' => $statusNoneMsg
+			],
+			'latestReviewed' => [
+				'statusKey' => self::STATUS_NONE,
+				'message' => $statusNoneMsg
+			],
+			'liveRevision' => [
+				'statusKey' => self::STATUS_NONE,
+				'message' => $statusNoneMsg
+			]
+		];
+	}
+
+	/**
+	 * Prepares data for given revision
+	 *
+	 * @param \Title $title
+	 * @param string $statusKey
+	 * @param int $revisionId
+	 * @param int $liveRevisionId
+	 * @return array
+	 */
+	private function prepareRevisionData( \Title $title, $statusKey, $revisionId, $liveRevisionId = 0 ) {
+		$data = [
+			'revisionId' => $revisionId,
+			'statusKey' => $statusKey,
+			'message' => $this->getStatusMessage( $statusKey ),
+			'diffLink' => '',
+			'reasonLink' => ''
+		];
+
+		if ( !empty( $revisionId ) ) {
+			$data['diffLink'] = $this->createRevisionLink( $title, $revisionId, $liveRevisionId );
+			if ( $statusKey === self::STATUS_REJECTED ) {
+				$data['reasonLink'] = $this->createRevisionTalkpageLink( $title );
+			}
+		} elseif ( !empty( $liveRevisionId ) ) {
+			$data['diffLink'] = $this->createRevisionLink( $title, $liveRevisionId );
+		}
+
+		return $data;
+	}
+
+	public function createPageLink( \Title $title ) {
+		return \HTML::element( 'a', [ 'href' => $title->getFullURL() ], $title->getText() );
```

Comment:
This will cause a fatal error:
`PHP Fatal error:  Class 'HTML' not found in extensions/wikia/ContentReview/services/ContentReviewStatusesService.class.php on line 297`

The class name is `Html` and our autoloader is case-sensitive, so same issue could happen in the other uses of the `Html` class below